packages/story-editor/src/components/canvas/karma/canvasKeyboardNavigation.karma.js

```diff
@@ -0,0 +1,147 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { Fixture } from '../../../karma';
+import { useStory } from '../../../app/story';
+
+describe('Canvas - keyboard navigation', () => {
+  let fixture;
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+
+    await fixture.render();
+    await fixture.collapseHelpCenter();
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  it('should not focus the canvas while tabbing through the editor', async () => {
+    const focusContainer = fixture.screen.getByTestId('canvas-focus-container');
+
+    // start focused on media pane searchbar
+    await fixture.events.focus(fixture.editor.library.media.searchBar);
+
+    // tab until focus reaches the canvas container
+    let count = 0;
+    while (count < 15) {
+      // eslint-disable-next-line no-await-in-loop -- need to await key press
+      await fixture.events.keyboard.press('tab');
+
+      if (document.activeElement === focusContainer) {
+        break;
+      }
+
+      count++;
+    }
+
+    if (count >= 15) {
+      throw new Error('Could not find focus container.');
+    }
+
+    // tab once more
+    await fixture.events.keyboard.press('tab');
+
+    // verify that prev page button is focused
+    expect(document.activeElement).toBe(
+      fixture.editor.canvas.framesLayer.prevPage
+    );
+  });
+
+  it('should focus the canvas and elements in the canvas after entering into the canvas space', async () => {
+    // add some elements
+    // add text to canvas
+    await fixture.editor.library.textTab.click();
+    await fixture.events.click(fixture.editor.library.text.preset('Paragraph'));
+    // add shape to canvas
+    await fixture.editor.library.shapesTab.click();
+    await fixture.events.click(fixture.editor.library.shapes.shape('Triangle'));
+    // add image to canvas
+    await fixture.editor.library.mediaTab.click();
+    await fixture.events.mouse.clickOn(
+      fixture.editor.library.media.item(0),
+      20,
+      20
+    );
+
+    // start focused on media pane searchbar
+    const focusContainer = fixture.screen.getByTestId('canvas-focus-container');
+
+    // start focused on media pane searchbar
+    await fixture.events.focus(fixture.editor.library.media.searchBar);
+
+    // tab until focus reaches the canvas container
+    let count = 0;
+    while (count < 15) {
+      // eslint-disable-next-line no-await-in-loop -- need to await key press
+      await fixture.events.keyboard.press('tab');
+
+      if (document.activeElement === focusContainer) {
+        break;
+      }
+
+      count++;
+    }
+
+    if (count >= 15) {
+      throw new Error('Could not find focus container.');
+    }
```

Comment:
Nit: We never pulled this into shared func. But I can do that with #11000

Reply:
Oh good catch! I'll do that real quick